openssl/src/asn1.rs

```diff
@@ -67,12 +67,16 @@ foreign_type_and_impl_send_sync! {
 impl fmt::Display for Asn1GeneralizedTimeRef {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         unsafe {
-            let mem_bio = MemBio::new()?;
-            cvt(ffi::ASN1_GENERALIZEDTIME_print(
-                mem_bio.as_ptr(),
-                self.as_ptr(),
-            ))?;
-            write!(f, "{}", str::from_utf8_unchecked(mem_bio.get_buf()))
+            match MemBio::new() {
```

Comment:
Can you split this up into multiple statements rather than one nested expression? You could also just propagate errors upstream, though I guess it makes sense to avoid failing the whole Display call.